### DISCUSSION TITLE
What is 'rank error' of aggregate functions?

### DISCUSSION THREAD

**Discussion by linter-purgatory** [prestodb/presto#16264] [Q&A]
We are looking at using `approx_percentile` to compute some statistics. I know that underneath those functions are implemented with the [q-digest data structure](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.132.7343) and have read [some resources that attempt to characterize it](https://skeptric.com/presto-approx-percentile/), but I still don't have a very good understanding of what 'rank error' is without going too into the weeds/implementation.

Could anyone help me understand what 'rank error' is, or point me to something that explains it? 

Appreciate the help!

**Comment by rongrong** [comment]
Can you share a failure stack trace?

**Reply by linter-purgatory** [reply]
It isn't a crash, it's understanding what 'rank error' means in the context of `approx_percentile`. That docs for that function says:

> As approx_percentile(x, percentage), but with a maximum rank error of accuracy. The value of accuracy must be between zero and one (exclusive) and must be constant for all input rows. Note that a lower “accuracy” is really a lower error threshold, and thus more accurate. The default accuracy is 0.01.

I'm trying to understand what they mean by 'rank error'
